internal/provider/log_export_resource.go

```diff
@@ -0,0 +1,372 @@
+				Attributes: map[string]schema.Attribute{
+					"api_key": schema.StringAttribute{
+						Required:            true,
+						MarkdownDescription: "Datadog API Key",
+					},
+					"datadog_host": schema.StringAttribute{
+						Required:            true,
+						MarkdownDescription: "Datadog site. Make sure to specify the correct site because Datadog sites are independent and data isn't shared across them by default",
+						Validators:          []validator.String{protoEnumValidator(logs.LogsTargetDatadog_DatadogHost_name)},
+					},
+				},
+				Validators: []validator.Object{},
+			},
+		},
+	}
+}
+
+func (l *LogExportResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	sdk, ok := req.ProviderData.(*dcsdk.SDK)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *dcsdk.SDK, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	l.sdk = sdk
+	l.logsExportService = l.sdk.Logs().Export()
+}
+
+func (l *LogExportResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *LogsExportResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createReq := &logs.CreateExportRequest{
+		ProjectId:   data.ProjectID.ValueString(),
+		Name:        data.Name.ValueString(),
+		Description: data.Description.ValueString(),
+		Sources:     nil,
+		Target:      nil,
+	}
+	for _, s := range data.Sources {
+		createReq.Sources = append(createReq.Sources, &logs.LogSource{
+			Type: logs.LogSourceType(logs.LogSourceType_value[s.Type.ValueString()]),
+			Id:   s.ID.ValueString(),
+		})
+	}
+	switch {
+	case data.S3 != nil:
+		createReq.Target = &logs.LogsTarget{Target: &logs.LogsTarget_S3{S3: &logs.LogsTargetS3{
+			Bucket:             data.S3.Bucket.ValueString(),
+			BucketLayout:       data.S3.BucketLayout.ValueString(),
+			AwsAccessKeyId:     data.S3.AWSAccessKeyID.ValueString(),
+			AwsSecretAccessKey: data.S3.AWSSecretAccessKey.ValueString(),
+			Region:             data.S3.Region.ValueString(),
+			Endpoint:           data.S3.Endpoint.ValueString(),
+			DisableSsl:         data.S3.DisableSSL.ValueBool(),
+			SkipVerifySslCert:  data.S3.SkipVerifySSLCert.ValueBool(),
+		}}}
+	case data.Datadog != nil:
+		createReq.Target = &logs.LogsTarget{Target: &logs.LogsTarget_Datadog{Datadog: &logs.LogsTargetDatadog{
+			ApiKey:      data.Datadog.APIKey.ValueString(),
+			DatadogHost: logs.LogsTargetDatadog_DatadogHost(logs.LogsTargetDatadog_DatadogHost_value[data.Datadog.DatadogHost.ValueString()]),
+		}}}
+	default:
+		resp.Diagnostics.AddError("misconfiguration", "at least one of \"s3\" or \"datadog\" must be specified")
```

Comment:
Maybe it's better to check it in plan modifier and fail during plan instead of apply?